---
title: "Sample 1"
author: "Matt Bixley"
date: "`r Sys.Date()`"
output:
  html_document: default
bibliography: ../data/r4biochem.bib
csl: ../data/pnas.csl
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE, 
                      fig.height = 10, fig.align = "center")
#output_dir <- "output"
library(tidyverse)

# plot themes
my_theme <- theme_classic() + 
  theme(legend.position="bottom") +
  theme(axis.line = element_line(colour = "grey50")) + # plot axis to grey
  theme(panel.grid.major = element_line(colour = "grey40", size = 0.4)) + # grid to grey
  theme(panel.grid.minor = element_line(colour = "grey70", size = 0.3)) + # grid to grey
  theme(panel.grid.major.x = element_blank()) + #remove the vertical grid lines 
  theme(panel.grid.minor.x = element_blank()) #remove the vertical grid lines

```

# Import Data

Lets read in a files and see how that looks.
```{r}
rawcount <- read_csv("../data/sample1.csv")
head(rawcount)
dim(rawcount)
```

OK that looks like we expected so maybe we have lots of them and want to create a large dataset

```{r loopread}
files <- c("sample1.csv", "sample2.csv") # list.files()

# create an empty file
gene_dat <- data.frame(matrix(ncol = ncol(rawcount))) 
names(gene_dat) <- colnames(rawcount)

for(file in files){
  x <- read_csv(paste0("../data/",file))
  gene_dat <- bind_rows(gene_dat, x)
}

gene_dat <- gene_dat[-1,]

# check the dimensions
dim(gene_dat)
head(gene_dat)
```

# Tidy

```{r tidy}

## Tidying
gene_final <- gene_dat %>% 
  # separate the sgRNA column into multiple new columns based on the "_" character as a delimiter 
    separate(col = sgRNA, into = c("something","TSS", "strand", "probe_gene"), sep = "_") %>%
  
  # separate the column that has the probe and gene info into two columns based on the "-" character
    separate(col = probe_gene, into = c("probe", "name"), sep = "-") %>% 
  
  # remove wording from TSS
    mutate(TSS = str_remove(TSS, pattern = "unique:")) %>% 
  
  ## mutate
  # Calculate fold change/ probe by using T20/T0
  mutate(fold_change = T20 / T0)

head(gene_final)

```



```{r join}
# join our probe data to the output
probe <- read_csv("../data/probes.csv")

gene_final <- gene_final %>% 
  left_join(.,probe) %>% 
  select(-something)

head(gene_final)

```

we sent `r length(files)` samples to [Genomics Aotearoa](https://www.genomics-aotearoa.org.nz/) for expression profiles from `r length(unique(gene_final$gene))` genes. the file is returned with the following column names ```r colnames(rawcount)```. We used the methods from [@breitling2004] in our analysis and maybe use the bioconductor package [@gentleman2004]

Table of Gene Summary Data
```{r genetable}
## gene summary
gene_summary <- gene_final %>%
    group_by(gene) %>%
    summarise(mean = mean(fold_change, na.rm = T), 
              sd = sd(fold_change, na.rm = T),
              n_probes = n())

knitr::kable(gene_summary, 
             digits = 3, # number of digits
             align = "lccc", # column alignment
             caption = "Summary fold change"
             )
```

Plot Fold Change by Gene

```{r geneplot, echo = FALSE, eval = FALSE}
p <- ggplot(gene_final, aes(x = gene, y = fold_change)) +
  geom_boxplot()
  #geom_violin(alpha = 0) +
  #geom_jitter(alpha = 0.6, width = 0.1, color = "blue")

p
```

```{r geneplotnice, echo = FALSE }
p1 <- ggplot(gene_final, aes(x = gene, y = fold_change)) +
  #geom_boxplot()
  geom_violin(alpha = 0) +
  geom_jitter(alpha = 0.6, width = 0.1, color = "blue") #+
  
p1 + my_theme + 
  labs(title = "Distribution of Fold Change\nby Gene",
       x = "Fold Change",
       y = "Gene Name") +
  theme(axis.text.x  = element_text(angle=45, vjust=0.6))

```

```{r nameplot, echo = FALSE, eval = F }
p2 <- ggplot(gene_final, aes(x = name, y = fold_change)) +
  #geom_boxplot()
  geom_violin(alpha = 0) +
  geom_jitter(alpha = 0.6, width = 0.1, color = "blue") +
  labs(title = "Distribution of Fold Change\nby Gene",
       x = "Fold Change",
       y = "Gene Name") +
  theme(axis.text.x  = element_text(angle=45, vjust=0.6))
  

p2
```


# Discussion


```{r bestgene}
best <- gene_summary %>% 
  filter(n_probes == max(n_probes)) # filter to the gene with most probes
```

For no better reason than it's a tidy looking name and has the most probes (n = `r best$n_probes`), we are interested in the gene **`r best$gene`** 

# References
